### ISSUE TITLE
Possible typo in LSGAN loss

### ISSUE THREAD

**Issue by ngoyal2707** [akanimax/pro_gan_pytorch#13]
The current code of [LSGAN](https://github.com/akanimax/pro_gan_pytorch/blob/master/pro_gan_pytorch/Losses.py#L175) loss does following:

```
0.5 * ((th.mean(self.dis(fake_samps, height, alpha)) - 1) ** 2)
```
which is `square of mean` of error, probably should be `mean of squared error` as following:
```
0.5 * (th.mean((self.dis(fake_samps, height, alpha) - 1) ** 2))
```

If above is correct, same applies to both `generator` and `discriminator` of `LSGAN` and `LSGANSigmoid`.
 
Note: I haven't actually run the above code, just noticed it while looking at the other losses in your code.

**Comment by akanimax** [comment]
@ngoyal2707,

Thank you for pointing out. Even I had never used the LSGAN with ProGAN as the paper had mentioned some additional stabilization tricks involving noise addition. Will make the required correction soon.

Best regards,
@akanimax

**Comment by ngoyal2707** [comment]
@akanimax Sure, I was using LSGAN in ProGAN with noise trick from paper which is giving decent result but sometimes still get random mode collapses where suddenly after some epochs, Generator starts producing random noise.

**Comment by akanimax** [comment]
@ngoyal2707, 
Interesting observation. I'll try it sometime too.
BTW, the error is fixed in [0368a2665b52b843785ffe0b18b3154fd32539b0](https://github.com/akanimax/pro_gan_pytorch/commit/0368a2665b52b843785ffe0b18b3154fd32539b0).

Closing the issue now.
Thanks once again.

Cheers :beers:!
@akanimax
